build/tasks/AddArchiveReferences.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Versioning;
+using RepoTasks.Utilities;
+
+namespace RepoTasks
+{
+    public class AddArchiveReferences : Task
+    {
+        [Required]
+        public string ReferencePackagePath { get; set; }
+
+        [Required]
+        public string RemoveTimestamp { get; set; } = "false";
```

Comment:
Nit: make this a bool. MSBuild will do the conversion for you. `public bool RemoveTimestamp`

Reply:
Ah I wasn't sure if that worked.